postings.go

```diff
@@ -158,24 +154,31 @@ func Merge(its ...Postings) Postings {
 }
 
 type mergePostings struct {
-	a, b     Postings
-	aok, bok bool
-	cur      uint32
+	a, b        Postings
+	initialized bool
+	aok, bok    bool
+	cur         uint32
 }
 
 func newMergePostings(a, b Postings) *mergePostings {
-	it := &mergePostings{a: a, b: b}
-	it.aok = it.a.Next()
-	it.bok = it.b.Next()
-
-	return it
+	return &mergePostings{a: a, b: b, aok: true, bok: true}
 }
 
 func (it *mergePostings) At() uint32 {
 	return it.cur
 }
 
 func (it *mergePostings) Next() bool {
+	if !it.initialized {
+		if !it.a.Next() {
+			it.aok = false
+		}
+		if !it.b.Next() {
+			it.bok = false
+		}
+		it.initialized = true
+	}
```

Comment:
Ah, I get it now.

> And yes, also wondering why the additional attribute instead of keeping the initialization in the constructor. I'm not sure I know which part of your other response you meant.

I meant my response to pt. 1 in this comment:
https://github.com/prometheus/tsdb/pull/57#issuecomment-296604719
(passing single postings to many other constructors, just to calculate it's cost).